From b0973874f835bf959c5d56ed55d325f35d5a1344 Mon Sep 17 00:00:00 2001
From: Justin DeMartino <jjdemartino@google.com>
Date: Thu, 7 Mar 2019 10:11:03 -0800
Subject: [PATCH 2/3] Fix Build Credentials refactored to use byte[]

 Conflicts:
	services/core/java/com/android/server/locksettings/LockSettingsService.java

[wight554: updated code from LA.QSSI.11.0.r1-05600-qssi.0]

Bug: 127810705
Change-Id: I766b553aef0479c286c0eee7302bb5e3b04b85bf
Signed-off-by: Volodymyr Zhdanov <wight554@gmail.com>
---
 .../com/android/server/locksettings/LockSettingsService.java  | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/locksettings/LockSettingsService.java b/services/core/java/com/android/server/locksettings/LockSettingsService.java
index ca8ab1b073d0..b66c82b3e56c 100644
--- a/services/core/java/com/android/server/locksettings/LockSettingsService.java
+++ b/services/core/java/com/android/server/locksettings/LockSettingsService.java
@@ -2197,7 +2197,9 @@ public class LockSettingsService extends ILockSettings.Stub {
                                             progressCallback, 0 /* flags */);
             if ((response.getResponseCode() == VerifyCredentialResponse.RESPONSE_OK) &&
                                             (userId == UserHandle.USER_OWNER)) {
-                    retainPassword(credential);
+                    //TODO(b/127810705): Update to credentials to use byte[]
+                    String credentialString = credential.isNone() ? null : new String(credential.getCredential());
+                    retainPassword(credentialString);
             }
             return response;
         } finally {
-- 
2.30.2

